configuration/passport.js

```diff
@@ -0,0 +1,38 @@
+const passport = require( 'passport' )
+const LocalStrategy = require( 'passport-local' ).Strategy
+
+const User = require( '../database/user' )
+
+const strategy = new LocalStrategy({
+    usernameField: 'email',
+    session: false,
+    passReqToCallback: true
+}, ( request, email, password, done ) => {
+  User.findByEmail( email )
+    .then( user => {
+      if( User.validPassword( password,  user.password ) ) {
+        return done( null, user )
+      } else {
+        return done( null, false, request.flash( 'message', 'Oops! Wrong password.' ))
+      }
+    })
+    .catch( error => {
+      return done( null, false, request.flash( 'message', 'Seems we don\'t have your email...' ) )
```

Comment:
Ooooo, gotcha. Should we just do this by writing in some conditional error handling?